esp32-wroom-rp/src/protocol.rs

```diff
@@ -13,9 +13,18 @@ use heapless::{String, Vec};
 
 use super::network::{ConnectionState, IpAddress, Port, Socket, TransportMode};
 use super::wifi::ConnectionStatus;
-use super::{Error, FirmwareVersion, ARRAY_LENGTH_PLACEHOLDER};
+use super::{Error, FirmwareVersion};
 
-pub(crate) const MAX_NINA_PARAM_LENGTH: usize = 4096;
+// The maximum number of NINA param u8 bytes in a command send/receive byte stream
+pub(crate) const MAX_NINA_PARAMS: usize = 8;
+
+pub(crate) const MAX_NINA_BYTE_PARAM_BUFFER_LENGTH: usize = 1;
+pub(crate) const MAX_NINA_WORD_PARAM_BUFFER_LENGTH: usize = 2;
+pub(crate) const MAX_NINA_SMALL_ARRAY_PARAM_BUFFER_LENGTH: usize = 255;
+pub(crate) const MAX_NINA_LARGE_ARRAY_PARAM_BUFFER_LENGTH: usize = 1024;
+
+// The maximum length that a 2-byte length NINA response can be
+pub(crate) const MAX_NINA_RESPONSE_LENGTH: usize = 32;
```

Comment:
I'm curious as to why `32` here? The way I read this is that we will only ever expect <= 32 bytes of data in responses. Is the intention here that we will increase this number when we start expecting larger responses? Ex. HTML Documents?